blocks.go

```diff
@@ -68,6 +66,13 @@ func (b *blocks) FinalizedHeader() (*flowgo.Header, error) {
 }
 
 func (b *blocks) ByHeightFrom(height uint64, header *flowgo.Header) (*flowgo.Header, error) {
-	return environment.NewBlockFinder(b.headers).
-		ByHeightFrom(height, header)
+	if height > header.Height {
+		return nil, storage.ErrNotFound
+	}
```

Comment:
I think you can also add

```go
if header.Height == height {
    return header, nil
}
```

Reply:
I think this one is in flow-go , but I will verify

Reply:
ah, didn't see that. Good catch